### PR TITLE
Feature/response lag

### PR DESCRIPTION
**Pull request by mllemango**
#### Link to the issue: 
https://github.com/de-souza-capstone-2020/Mars/issues/56

#### Description of Pull Request:
Added response lag and typing indicator

#### Test Plan:

- [x] When selecting a response the chatbot will take 2 seconds before replying
- [x] there should be an indicator "chatbot is typing..." during the pause
- [x] typing indicator should disappear when new reply is sent

### CODE REVIEW

**Comment by mllemango** [comment]
Screenshot_1579134177

File: screens/home.js [hunk 1]
```diff
@@ -1,208 +1,222 @@
...
+  };
+  renderQuickReplySend = () => <Text>{" custom send =>"}</Text>;
+  renderInputToolbar(props) {
+    if (this.state.toolbar) {
+      return <InputToolbar {...props} />;
     }
-    render() {
+  }
+  renderFooter = () => {
+    if (this.state.typingText) {
```
Comment (itsDZhang) [inline screens/home.js]: Tends to be bad practice for `this.state.<something>` if that "something" occurs more than once. 
Usually it'll be like:
`const { typingText } from this.state;`
That way, it reduces the likelihood of the entire file flooded with `this.state`.
Change in the future if you want but we're not going to be that strict here.

File: screens/home.js [hunk 2]
```diff
@@ -1,208 +1,222 @@
...
+    if (replies.length === 1) {
+      this.onSend([
+        {
+          createdAt,
+          _id: Math.round(Math.random() * 1000000),
+          text: replies[0].title,
+          user
+        }
+      ]);
+      setTimeout(() => this.determineResponse(replies[0]), 2000);
```
Comment (itsDZhang) [inline screens/home.js]: Although I think 2 seconds is a bit long as a delay will reduce user experience. Maybe we can decrease it in the future? I think 1 second is a good enough wait time. Up to you

Comment (mllemango) [reply screens/home.js]: i agree, lets play around with this value in the future

**Pull request merged**